### PR TITLE
Add basic template.

### PR DESCRIPTION
**Pull request by aliihsancengiz**
Related https://github.com/Men-Digital-Solutions/Men-Digital-Solutions.github.io/issues/1

### CODE REVIEW

**Comment by aliihsancengiz** [comment]
Utility scripts and sanitizers will be added on different pr.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Event Bus Pipeline
```
Comment (hmenn) [inline .github/workflows/ci.yml]: please change name

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.15)
+
+set(PROJECT_NAME template-project)
+set(PROJECT_VERSION 1.0.0)
```
Comment (hmenn) [inline CMakeLists.txt]: no need to define specific variable for project_version. 

project(TEMPLATE_PROJECT
        VERSION 1.0.0
        LANGUAGES CXX)

after this definition, you're able to access version via PROJECT_VERSION. (https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html)

same is also applicable for PROJECT_NAME

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+option(BUILD_TEST "Build with test" OFF)
+
+add_subdirectory(Src)
+
+if(BUILD_TEST)
+	add_subdirectory(Test)
```
Comment (hmenn) [inline CMakeLists.txt]: do not use tabs here. Always use spaces and for cmake it should be 4.

File: Inc/utils.hpp [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (hmenn) [inline Inc/utils.hpp]: please start folder names with lower cases.

Inc, Src, Test -> inc, src, test

File: Src/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	${CMAKE_CURRENT_SOURCE_DIR}/utils.cpp
+)
+
+add_executable(${TARGET_NAME} ${TARGET_SOURCES})
+
+target_include_directories(${TARGET_NAME}
+	PUBLIC
+		${PROJECT_SOURCE_DIR}/Inc
+	PRIVATE
+
```
Comment (hmenn) [inline Src/CMakeLists.txt]: follow space and tab rules

File: Src/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+target_include_directories(${TARGET_NAME}
+	PUBLIC
+		${PROJECT_SOURCE_DIR}/Inc
+	PRIVATE
+	
+)
+
+target_link_libraries(${TARGET_NAME}
+	PUBLIC
+
```
Comment (hmenn) [inline Src/CMakeLists.txt]: if it's empt, at least add ""

File: Src/main.cpp [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+
+int main()
+{
+	std::cout << "Hello World" << std::endl;
```
Comment (hmenn) [inline Src/main.cpp]: no tabs in code.  Generally no tab in project.

File: Src/main.cpp [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+
+int main()
```
Comment (hmenn) [inline Src/main.cpp]: add command line arguments as well.

Comment (hmenn) [reply Src/main.cpp]:
```suggestion
int main(int argc, char *argv[])
```

File: Src/utils.cpp [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#include "utils.hpp"
+
+namespace common_utils {
```
Comment (hmenn) [inline Src/utils.cpp]: first namespace should be project namespace.
```suggestion
namespace template_project::common_utils {
```

File: Test/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,22 @@
+set(TARGET_NAME "run_tests")
+
+set(TEST_SRC
+	TemplateTests.cpp
```
Comment (hmenn) [inline Test/CMakeLists.txt]: filename should not start with capital case

File: Test/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+find_package(GTest REQUIRED)
+find_package(Threads REQUIRED)
+
+add_executable(${TARGET_NAME} ${TEST_SRC})
+include_directories(${PROJECT_SOURCE_DIR}/include)
+
+target_link_libraries(${TARGET_NAME}
+	PUBLIC
+
+	PRIVATE
```
Comment (hmenn) [inline Test/CMakeLists.txt]:
```suggestion
     PUBLIC
         ""
     PRIVATE
```

**Review by hmenn — CHANGES_REQUESTED** [review]

**Review by hmenn — APPROVED** [review]

**Pull request merged**